### PR TITLE
Update docs and examples and tests to use NewClient instead of Dial

### PR DESCRIPTION
**Pull request by silaselisha**
- [x] refactored tests and examples that utilized `grpc.Dial` to `grpc.NewClient` while some tests were left to use the deprecated `grpc.Dial`

RELEASE NOTES: none

### CODE REVIEW

**Comment by arvindbr8** [comment]
Nice work on the PR. Thanks for taking a look at this. However while grep-ing, I still see more usages of `grpc.Dial`. I think we should probably add all of that as part of this patchset

<details>
<summary>grep output</summary>

```
./Documentation/encoding.md:	myclient := grpc.Dial(ctx, target, grpc.WithDefaultCallOptions(grpc.CallContentSubtype("mycodec")))
./Documentation/encoding.md:	myclient := grpc.Dial(ctx, target, grpc.WithDefaultCallOptions(grpc.UseCompressor("gzip")))
./Documentation/grpc-auth-support.md:conn, err := grpc.Dial(serverAddr, grpc.WithTransportCredentials(credentials.NewClientTLSFromCert(nil, "")))
./Documentation/grpc-auth-support.md:conn, err := grpc.Dial(serverAddr, grpc.WithTransportCredentials(credentials.NewClientTLSFromCert(nil, "")), grpc.WithPerRPCCredentials(oauth.NewComputeEngine()))
./Documentation/grpc-auth-support.md:conn, err := grpc.Dial(serverAddr, grpc.WithTransportCredentials(credentials.NewClientTLSFromCert(nil, "")), grpc.WithPerRPCCredentials(jwtCreds))
./resolver_balancer_ext_test.go:	cc, err := grpc.Dial(name+":///", grpc.WithTransportCredentials(insecure.NewCredentials()))
./resolver_balancer_ext_test.go:	cc, err := grpc.Dial(name+":///", grpc.WithTransportCredentials(insecure.NewCredentials()))
./test/local_creds_test.go:		cc, err = grpc.Dial(lisAddr, grpc.WithTransportCredentials(local.NewCredentials()), grpc.WithContextDialer(
./test/creds_test.go:	cc, err := grpc.Dial(r.Scheme()+":///"+testAuthority, grpc.WithTransportCredentials(cred), grpc.WithResolvers(r))
./test/creds_test.go:	cc, err := grpc.Dial(r.Scheme()+":///"+testAuthority, grpc.WithTransportCredentials(cred), grpc.WithResolvers(r))
./test/creds_test.go:	cc, err := grpc.Dial(lis.Addr().String(), grpc.WithTransportCredentials(cred))
./test/balancer_switching_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./test/balancer_switching_test.go:	cc, err := grpc.Dial(target, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./test/balancer_switching_test.go:	cc, err := grpc.Dial(target, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./test/balancer_switching_test.go:	cc, err := grpc.Dial(target, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./test/balancer_switching_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./test/balancer_switching_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./test/balancer_switching_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./test/goaway_test.go:	cc, err := grpc.Dial(lis.Addr().String(), grpc.WithTransportCredentials(insecure.NewCredentials()))
./test/clientconn_state_transition_test.go:	client, err := grpc.Dial("",
./test/clientconn_state_transition_test.go:	client, err := grpc.Dial(lis.Addr().String(),
./test/clientconn_state_transition_test.go:	client, err := grpc.Dial("whatever:///this-gets-overwritten",
./test/clientconn_state_transition_test.go:	client, err := grpc.Dial("whatever:///this-gets-overwritten",
./test/clientconn_state_transition_test.go:	cc, err := grpc.Dial(testResName+":///",
./test/authority_test.go:	cc, err := grpc.Dial(":"+port, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
./test/roundrobin_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", dopts...)
./test/pickfirst_test.go:	cc, err := grpc.Dial(lis.Addr().String(), dopts...)
./test/pickfirst_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", dopts...)
./test/pickfirst_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", dopts...)
./test/end2end_test.go:	te.cc, err = grpc.Dial(scheme+te.srvAddr, opts...)
./test/end2end_test.go:	te.cc, err = grpc.Dial(scheme+te.srvAddr, opts...)
./test/end2end_test.go:	cc, err := grpc.Dial(lis.Addr().String(), grpc.WithTransportCredentials(insecure.NewCredentials()))
./test/end2end_test.go:	cc, err := grpc.Dial("passthrough:///"+lis.Addr().String(), grpc.WithTransportCredentials(&clientFailCreds{}))
./test/end2end_test.go:	cc, err := grpc.Dial(lis.Addr().String(), grpc.WithTransportCredentials(insecure.NewCredentials()))
./test/end2end_test.go:	cc, err := grpc.Dial(lis.Addr().String(), grpc.WithTransportCredentials(insecure.NewCredentials()))
./test/end2end_test.go:	cc, err := grpc.Dial(mr.Scheme()+":///", grpc.WithResolvers(mr), grpc.WithStatsHandler(sh), grpc.WithTransportCredentials(insecure.NewCredentials()))
./test/healthcheck_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", opts...)
./test/xds/xds_client_federation_test.go:	_, err = grpc.Dial(target, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(resolver))
./test/xds/xds_client_certificate_providers_test.go:	cc, err := grpc.Dial(fmt.Sprintf("xds:///%s", serviceName), grpc.WithTransportCredentials(creds), grpc.WithResolvers(resolverBuilder))
./test/resolver_update_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./test/resolver_update_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./dialoptions.go:// security is incompatible and will cause grpc.Dial() to fail.
./interop/grpclb_fallback/client_linux.go:	conn, err := grpc.Dial(*serverURI, opts...)
./interop/test_utils.go:		conn, err = grpc.Dial(serverAddr, dopts...)
./interop/observability/client/client.go:	conn, err := grpc.Dial(serverAddr, grpc.WithTransportCredentials(insecure.NewCredentials()))
./interop/http2/negative_http2_client.go:	conn, err := grpc.Dial(serverAddr, opts...)
./interop/alts/client/client.go:	conn, err := grpc.Dial(*serverAddr, grpc.WithTransportCredentials(altsTC), grpc.WithBlock())
./interop/xds_federation/client.go:		cc, err := grpc.Dial(uris[i], opts...)
./interop/xds/client/client.go:		conn, err := grpc.Dial(*server, grpc.WithTransportCredentials(creds))
./interop/client/client.go:	conn, err := grpc.Dial(serverAddr, opts...)
./interop/stress/metrics_client/main.go:	conn, err := grpc.Dial(*metricsServerAddress, grpc.WithTransportCredentials(insecure.NewCredentials()))
./interop/stress/client/main.go:	return grpc.Dial(address, opts...)
./admin/test/utils.go:	conn, err := grpc.Dial(lis.Addr().String(), grpc.WithTransportCredentials(insecure.NewCredentials()))
./internal/idle/idle_e2e_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", dopts...)
./internal/idle/idle_e2e_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", dopts...)
./internal/idle/idle_e2e_test.go:			cc, err := grpc.Dial(r.Scheme()+":///test.server", dopts...)
./internal/idle/idle_e2e_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", dopts...)
./internal/idle/idle_e2e_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", dopts...)
./internal/stubserver/stubserver.go:	cc, err := grpc.Dial(ss.Target, opts...)
./balancer/rls/control_channel.go:	ctrlCh.cc, err = grpc.Dial(rlsServerName, dopts...)
./balancer/rls/balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///rls.test.example.com", grpc.WithResolvers(r), grpc.WithTransportCredentials(clientCreds))
./balancer/rls/balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///rls.test.example.com", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/picker_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/picker_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/picker_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/picker_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/picker_test.go:			cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/picker_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/picker_test.go:	cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/picker_test.go:			cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/picker_test.go:			cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/picker_test.go:			cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/picker_test.go:			cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/rls/picker_test.go:			cc, err := grpc.Dial(r.Scheme()+":///", grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./balancer/grpclb/grpclb_test.go:	cc, err := grpc.Dial(r.Scheme()+":///"+beServerName, dopts...)
./balancer/grpclb/grpclb_test.go:	cc, err := grpc.Dial(r.Scheme()+":///"+beServerName, dopts...)
./balancer/grpclb/grpclb_test.go:	cc, err := grpc.Dial(r.Scheme()+":///"+beServerName, dopts...)
./balancer/grpclb/grpclb_test.go:	cc, err := grpc.Dial(r.Scheme()+":///"+beServerName,
./balancer/grpclb/grpclb_test.go:	cc, err := grpc.Dial(r.Scheme()+":///"+beServerName,
./balancer/grpclb/grpclb_remote_balancer.go:	cc, err := grpc.Dial(target, dopts...)
./profiling/cmd/remote.go:	cc, err := grpc.Dial(*flagAddress, grpc.WithTransportCredentials(insecure.NewCredentials()))
./examples/gotutorial.md:To call service methods, we first need to create a gRPC *channel* to communicate with the server. We create this by passing the server address and port number to `grpc.Dial()` as follows:
./examples/gotutorial.md:conn, err := grpc.Dial(*serverAddr)
./examples/helloworld/greeter_client/main.go:	conn, err := grpc.Dial(*addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
./examples/features/retry/README.md:conn, err := grpc.Dial(ctx,grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithDefaultServiceConfig(retryPolicy))
./examples/features/health/README.md:conn, err := grpc.Dial(..., serviceConfig)
./examples/features/observability/client/main.go:	conn, err := grpc.Dial(*addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
./examples/route_guide/client/client.go:	conn, err := grpc.Dial(*serverAddr, opts...)
./xds/internal/test/e2e/e2e.go:	cc, err := grpc.Dial(fmt.Sprintf("localhost:%d", clientStatsPort), grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithDefaultCallOptions(grpc.WaitForReady(true)))
./xds/internal/balancer/clusterresolver/e2e_test/aggregate_cluster_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./xds/internal/balancer/clusterresolver/e2e_test/eds_impl_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./xds/internal/balancer/clusterresolver/e2e_test/eds_impl_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./xds/internal/balancer/clusterresolver/e2e_test/eds_impl_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./xds/internal/balancer/clusterresolver/e2e_test/eds_impl_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./xds/internal/balancer/clusterresolver/e2e_test/eds_impl_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./xds/internal/balancer/clusterresolver/e2e_test/eds_impl_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./xds/internal/balancer/clusterresolver/e2e_test/eds_impl_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./xds/internal/balancer/ringhash/e2e/ringhash_balancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.server", dopts...)
./xds/internal/balancer/cdsbalancer/cdsbalancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./xds/internal/balancer/cdsbalancer/cdsbalancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./xds/internal/balancer/cdsbalancer/cdsbalancer_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
./xds/internal/balancer/cdsbalancer/cdsbalancer_security_test.go:	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(clientCreds), grpc.WithResolvers(r))
./xds/googledirectpath/googlec2p_test.go:	cc, err := grpc.Dial(uri, grpc.WithTransportCredentials(insecure.NewCredentials()))
./stats/opencensus/opencensus.go:// grpc.Dial().
./stats/stats_test.go:	te.cc, err = grpc.Dial(te.srvAddr, opts...)
./resolver/manual/manual_test.go:		_, err := grpc.Dial("whatever://localhost",
./orca/producer_test.go:	cc, err := grpc.Dial("whatever:///whatever", grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"customLB":{}}]}`), grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./orca/producer_test.go:	cc, err := grpc.Dial("whatever:///whatever", grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"customLB":{}}]}`), grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./orca/producer_test.go:	cc, err := grpc.Dial("whatever:///whatever", grpc.WithDefaultServiceConfig(`{"loadBalancingConfig": [{"customLB":{}}]}`), grpc.WithResolvers(r), grpc.WithTransportCredentials(insecure.NewCredentials()))
./credentials/insecure/insecure.go:// transport security is incompatible and will cause grpc.Dial() to fail.

```
</details>

**Comment by silaselisha** [comment]
@arvindbr8 a polite request for clarity on `Testing, Attempt #2 / tests (extras, 1.22)`  failing. I'll be more than happy to redo the task.

**Comment by arvindbr8** [comment]
@silaselisha -- Thanks for bringing that issue to our notice. Seems like there is a bug in NewClient that is causing a regression. Investigating it [here](https://github.com/grpc/grpc-go/issues/7075)

**Comment by arvindbr8** [comment]
But that shouldn't necessarily block your PR

**Comment by silaselisha** [comment]
@arvindbr8 I'll look into issue your raised, to learn the dynamics of the `grpc.NewClient`

**Comment by arvindbr8** [comment]
I turned up the logs with `GRPC_GO_LOG_SEVERITY_LEVEL=info GRPC_GO_LOG_VERBOSITY_LEVEL=99` and noticed that the orcaLB that is passed in the test is asserting that the `len(addrs) == 1` here: https://github.com/grpc/grpc-go/blob/4ec8307379292dcbe4105ef65d6f9d1f569aa87d/examples/features/orca/client/main.go#L103-L106


In the case of the Github VM (and in my local machine) the dns resolver returned both the ipv6 and ipv4 address.

It's unclear why we need that since the example should work even if there are more addresses. I would just remove that conditional in `UpdateClientConnState` method in orcaLB

**Comment by silaselisha** [comment]
@arvindbr8 if both ipv4 and ipv6 are returned, is it possible to just change the condition to `len(addrs) != 2`? And are there instances where one `ip` version will be expected?

**Comment by arvindbr8** [comment]
Its not deterministic because it depends on the machine you run the test in and the ENI(s) configured. I would just get rid of the conditional there.

**Comment by silaselisha** [comment]
@arvindbr8 since `grpc.DialWithContext` is deprecated too, will it be right for it to be part of this PR?

**Comment by arvindbr8** [comment]
Sure. I'm also okay even if we decide to do this in increments.

**Comment by silaselisha** [comment]
@arvindbr8 a polite request you review this issue here.

**Comment by arvindbr8** [comment]
@silaselisha -- Seems like maybe you didnt push your branch to remote? I'm not seeing a new commit after the last review. I'm particular about this [comment](https://github.com/grpc/grpc-go/pull/7068#issuecomment-2033167289) which would help satisfy Testing (extras) /orca example

**Comment by silaselisha** [comment]
@arvindbr8 in reference to a closed issue #7075 that you had created to fix Testing (extras) /orca example.

**Comment by silaselisha** [comment]
> I turned up the logs with `GRPC_GO_LOG_SEVERITY_LEVEL=info GRPC_GO_LOG_VERBOSITY_LEVEL=99` and noticed that the orcaLB that is passed in the test is asserting that the `len(addrs) == 1` here:
> 
> https://github.com/grpc/grpc-go/blob/4ec8307379292dcbe4105ef65d6f9d1f569aa87d/examples/features/orca/client/main.go#L103-L106
> 
> In the case of the Github VM (and in my local machine) the dns resolver returned both the ipv6 and ipv4 address.
> 
> It's unclear why we need that since the example should work even if there are more addresses. I would just remove that conditional in `UpdateClientConnState` method in orcaLB

I would also like to refer you to this comment.

**Comment by arvindbr8** [comment]
I closed #7075 because the error was with the way the test was setup. We shouldnt ideally be expecting the resolver to only return 1 address. That's why I mentioned in the [comment](https://github.com/grpc/grpc-go/pull/7068#issuecomment-2033167289) that the fix for the test can be along with this PR. 

Ideally @dfawley mentioned how he would like to split the update in #7049. Your PR seems like it does tackle most of (1) and (2). This is better for tracking. 

If you don't prefer to do that, you can revert your changes to `examples/` and someone else could pick up that piece.

**Comment by silaselisha** [comment]
I don't mind including that piece in this PR, I am working on the missing piece, and I'll push the changes asp.

**Comment by silaselisha** [comment]
@arvindbr8 I mad the necessary changes but I now the PR it has more unnecessary commits

**Comment by dfawley** [comment]
Something is wrong with your fork's git repo, apparently.  If you can't figure it out, worst case: you can make a new directory and (assuming `upstream` is `github.com/grpc/grpc-go`): `get fetch upstream; git reset --hard upstream/master`, then `cp -rf` from the old directory into the new directory.

File: examples/features/orca/client/main.go
Comment (arvindbr8) [inline examples/features/orca/client/main.go:106]: nix these lines instead of comment? 

```suggestion
```

File: Documentation/anti-patterns.md
Comment (arvindbr8) [inline Documentation/anti-patterns.md:41]: Do we need to add this here since `WithBlock` is deprecated, and is not available in NewClient?

File: Documentation/anti-patterns.md
Comment (arvindbr8) [inline Documentation/anti-patterns.md:60]: I would prefer changing the content under `### Dialing in gRPC` to talk about `NewClient` and swap the current content under the heading over here.

File: Documentation/anti-patterns.md
Comment (arvindbr8) [inline Documentation/anti-patterns.md:61]: please line wrap to 80 cols. Applies to all the changes in this file.

File: Documentation/anti-patterns.md
Comment (arvindbr8) [inline Documentation/anti-patterns.md:67]:
```suggestion
`grpc.NewClient` uses passthrough as the default name resolver for backward compatibility while  `Dial` uses dns as its default name resolver. This subtle difference is crucial in legacy systems that specify a custom dialer and expect it to receive the target string directly.
```

File: Documentation/anti-patterns.md
Comment (arvindbr8) [inline Documentation/anti-patterns.md:65]:
```suggestion
`grpc.NewClient` automatically ignores `DialOptions` returned by `WithBlock`, `WithTimeout`, `WithReturnConnectionError`, and `FailOnNonTempDialError`. 
```

File: Documentation/anti-patterns.md
Comment (arvindbr8) [inline Documentation/anti-patterns.md:67]: `grpc.NewClient` uses `dns` and `grpc.Dial` uses `passthrough`

File: Documentation/anti-patterns.md
Comment (arvindbr8) [inline Documentation/anti-patterns.md:1]: I'm not fully convinced that this is the correct place for this documentation to go to. How about we revert the changes here, and someone internal could possibly pick up how the messaging should look like.

**Review by arvindbr8 — CHANGES_REQUESTED** [review]

**Comment by arvindbr8** [comment]
@silaselisha -- I've addressed my comments and pushed a new commit to your branch. I think it looks good now. Per policy we would want one more review on the PR from our maintainers. cc: @zasweq, could you please take a pass at the documentation update and the other diffs

**Review by arvindbr8 — APPROVED** [review]

**Comment by zasweq** [comment]
Giving to Abhishek since he's on call.

**Review by aranjans — APPROVED** [review]

**Comment by aranjans** [comment]
LGTM!

**Review by arvindbr8 — APPROVED** [review]

**Comment by arvindbr8** [comment]
Thanks for the PR @silaselisha.

**Pull request merged**
